rootex/core/audio/audio_source.cpp

```diff
@@ -73,6 +73,53 @@ ALuint AudioSource::getSourceID() const
 	return m_SourceID;
 }
 
+void AudioSource::setPosition(Vector3 position)
+{
+	float x = position.x;
+	float y = position.y;
+	float z = position.z;
+	AL_CHECK(alSource3f(m_SourceID, AL_POSITION, x, y, z));
+}
+
+void AudioSource::setRolloffFactor(ALfloat rolloffFactor)
+{
+	AL_CHECK(alSourcef(m_SourceID, AL_ROLLOFF_FACTOR, rolloffFactor));
+}
+
+void AudioSource::setReferenceDistance(ALfloat referenceDistance)
+{
+	AL_CHECK(alSourcef(m_SourceID, AL_REFERENCE_DISTANCE, referenceDistance));
+}
+
+void AudioSource::setMaxDistance(ALfloat maxDistance)
+{
+	AL_CHECK(alSourcef(m_SourceID, AL_MAX_DISTANCE, maxDistance));
+}
+
+void AudioSource::setModel(AudioSource::AttenuationModel model)
+{
+	switch (model)
```

Comment:
You will not need this here. Check the comment at enum class AttenuationModel

Reply:
Can you please elaborate?